### PR TITLE
[serve] Use public api for status and handle in tests

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Use new public api `serve.status()`, `serve.get_app_handle()`, `serve.get_deployment_handle()` in tests when appropriate.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/tests/test_autoscaling_policy.py [hunk 1]
```diff
@@ -1205,7 +1205,9 @@ def g():
     }
 
     client.deploy_apps(ServeDeploySchema(**{"applications": [app_config]}))
-    wait_for_condition(lambda: client.get_serve_status().app_status.status == "RUNNING")
+    wait_for_condition(
+        lambda: serve.status().applications["default"].status == "RUNNING"
```
Comment (GeneDer) [inline python/ray/serve/tests/test_autoscaling_policy.py:1209]: Non-blocker: Kinda nitpick to maybe switch all the usage of `"default"` to `SERVE_DEFAULT_APP_NAME` XP

**Review by GeneDer — APPROVED** [review]
Small non-blocker nitpick, thanks for replacing all the api calls!

File: python/ray/serve/tests/test_autoscaling_policy.py (hunk 1 above)
Comment (zcin) [reply python/ray/serve/tests/test_autoscaling_policy.py]: Good point, I've replaced the raw string with the constant!

**Review by edoakes — APPROVED** [review]
Awesome! 🙏

**Comment by edoakes** [comment]
@zcin serve tests failing though

**Comment by zcin** [comment]
@edoakes Tests are passing!

**Review by sihanwang41 — APPROVED** [review]

**Pull request merged**